pkg/cli/options.go

```diff
@@ -139,3 +150,150 @@ func WithCompletion() Option {
 		return nil
 	}
 }
+
+// parseExternalPluginArgs parses external plugin arguments and trims plugins flag and its values from args.
+func parseExternalPluginArgs() (extPlugins []string, args []string) {
+	var externalPlugins string
+	args = make([]string, len(os.Args)-1)
+	copy(args, os.Args[1:])
+
+	// parse os.Args and pass everything to plugin except program name,
+	// --plugins flag and its value.
+	for i, item := range args {
+		if item == "--plugins" {
+			externalPlugins = args[i+1]
+			args = append(args[:i], args[i+2:]...)
+		}
+	}
+
+	extPlugins = strings.Split(externalPlugins, ".")
+
+	return extPlugins, args
+}
+
+// setPluginsRootBasedonHost detects the host system and sets the plugins root based on the host.
+func setPluginsRootBasedonHost() (pluginsRoot string) {
```

Comment:
Nit: naming, and just return an error.
```suggestion
func getPluginsRoot() (pluginsRoot string, err error) {
```

Reply:
We are actually setting the pluginsRoot based on the host. But I am good with changing it to `getPluginsRoot()`.